src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Maintaining Subsystems
+description: How to maintain subsystems within the main codebase.
+slug: /guides/main/maintaining-subsystems
+---
+
+This page details how to maintain various subsystems within the main codebase.
+
+# Vision
+
+Before vision can work on the robot, odometry and kinematics must work on the robot. If these are inaccurate, then the green horizon will be inaccurate.
+
+## Dataset Generation
+
+Datasets for training the Visual Mesh can be found on the Network-Attached Storage (NAS). If you don't have access to the NAS, ask a team leader for an account. The NAS contains millions of NUpbr training images and large TensorFlow records containing the correct data to train the Visual Mesh on.
+
+### NUpbr
+
+NUpbr is a Physically Based Rendering tool created in Blender. It creates semi-synthetic images with corresponding segmentation images for training.
+
+NUpbr requires Blender **2.79**. Download the appropriate file from the [Blender 2.79 Stable Release](https://download.blender.org/release/Blender2.79/).
+
+Clone the [NUpbr repository](https://github.com/NUbots/NUpbr) using
+
+```sh
+git clone https://github.com/NUbots/NUpbr.git
+```
+
+Navigate to the NUpbr directory and run
+
+```sh
+blender --python pbr.py
```

Comment:
This seems platform specific to Linux/Macos/WSL, but it doesn't mention that anywhere. Would this work on normal windows as is? Maybe it would, with git bash? not sure though. Also, this makes an assumption that blender is in $PATH, which might(?) have to be done manually for macs (I have no idea about windows).